string ctrl_str
---
string ctrl_str
